.github/workflows/pressio_testing.yml

```diff
@@ -0,0 +1,39 @@
+name: Pressio Testing
+
+on:
+  push:
+    branches:
+      - CI-improvements-docker-images
+
+jobs:
+  Build-and-Run-tests:
+    strategy:
+      max-parallel: 10
+      fail-fast: false
+      matrix:
+        config:
+          - { os: ubuntu-latest, image: ubuntu-20.04-gnu_9-eigen_3.3.7-gtest }
+          - { os: ubuntu-latest, image: ubuntu-20.04-clang_9-eigen_3.3.7-gtest }
+#          - { os: ubuntu-latest, image: ubuntu-20.04-intel-eigen_3.3.7-gtest }
+          - { os: ubuntu-latest, image: fedora-34-gnu_11-eigen_3.3.7-gtest }
+          - { os: ubuntu-latest, image: fedora-34-clang_12-eigen_3.3.7-gtest }
+    runs-on: ${{ matrix.config.os }}
+    container: pressiomodelreduction/${{ matrix.config.image }}
+    env:
+      pressio_dir: /__w/pressio/pressio
+      target_dir: /out
+      eigen_path: /usr/local/eigen/install
+      gtest_path: /usr/local/gtest/install
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build pressio tests
+        run: |
+          cd /pressio-builder
+          git pull
+          git status
+          sed -i 's/source .\/shared\/colors.sh/# colors commnted out/g' main_pressio.sh
+          ./main_pressio.sh -dryrun=no -build-mode=Release -target-dir=${{env.target_dir}} -eigen-path=${{env.eigen_path}} -gtest-path=${{env.gtest_path}} -cmake-generator-name=default_with_tests -pressio-src=${{env.pressio_dir}}
```

Comment:
@marcinwrobel1986 before merging this, can you please add also builds for "debug" ? for all images, I would like both Release and Debug - thanks

Reply:
Ok, will do. I will let you know, when done.

Reply:
thanks! no rush!